core/include/dt-bindings/gpio/stm32mp_gpio.h

```diff
@@ -0,0 +1,14 @@
+/* SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-3-Clause) */
+/*
+ * Copyright (C) STMicroelectronics 2023 - All Rights Reserved
+ */
+
+#ifndef _DT_BINDINGS_GPIO_STM32MP_GPIO_H
+#define _DT_BINDINGS_GPIO_STM32MP_GPIO_H
+
+#include <dt-bindings/gpio/gpio.h>
+
+/* Macro to define the security for GPIO */
+#define TZPROT(id) (1 << (id))
```

Comment:
This can give unexpected results when `id` is 31 since `1` is signed, compare this with the BIT() macro.

Reply:
Hi Jens,
since this will be used in the device-tree I cannot use types or U().
Is there really a notion of signs in the device-tree files?

Reply:
I don't know, perhaps it's like assembly instead. Anyway, this macro is available to C code and as you can see in the BIT() macro implementation we deal with assembly too.

Reply:
I'm not an expert on device tree pre-processing but I guess there's a Cpp process to handle things such as includes. There's a usecase of I2C_TEN_BIT_ADDRESS that is (1 << 31) in `include/dt-bindings/i2c/i2c.h` that is then used by I2C OF core but is quite old.

FYI, STM32 GPIOs aren't expected to exceed 16 GPIOs per bank so id will stay < 16.

Reply:
Perhaps C compatibility isn't a thing for .h files consumed by device-tree files. But I find it odd to have then have them available in the include path for C files.

Reply:
I can ignore this if others think this is OK.

Reply:
`.dts` files are preprocessed with `$(CROSS_COMPILE) -x assembler-with-cpp ...` so they have `__ASSEMBLER__` defined. For what we want here this is enough, because it selects the proper form of the `BIT()` macro that `dtc` will understand. If some day we need more clever things we could add `-D__DTS__` to the preprocessor command (like Linux does) so we could differentiate even more between C, assembler and DT.

Reply:
Given that it isn't hard to do the right thing, please fix this.